### ISSUE TITLE
Add a warning sign for anomaly detection

### ISSUE THREAD

**Issue by prajjwal1** [pytorch/pytorch#33176]
## 🚀 Feature
Autograd's anomaly detect is used for debugging purposes, primarily to catch Nans, it would be good to add a warning sign to denote that it is enabled by the user and would increase the training time. This can prevent unintended enablement of `detect_anomaly`. You can ignore this request as it is not a priority, I can open a PR for this if you like. It'll be good to have a warning sign. 

cc @ezyang @SsnL @albanD @zou3519 @gqchen

**Comment by albanD** [comment]
Hi,

That sounds like a good idea. 
We would be happy to accept a PR for that !

**Comment by ezyang** [comment]
@albanD Shouldn't we merge "actionable" into "proposal accepted" label? Or are these different things?

**Comment by albanD** [comment]
@prajjwal1 note that the python warning package should be used to do this. Just like it is done [here](https://github.com/pytorch/pytorch/blob/1767ae8daf48caf7514e565a956092aec3105cf5/torch/tensor.py#L365).

**Comment by prajjwal1** [comment]
@albanD Thanks for pointing about the warnings package. I've done it [here](https://github.com/pytorch/pytorch/pull/33239). Please let me know if something doesn't work.

**Comment by prajjwal1** [comment]
Closing this.
